### ISSUE TITLE
Consider disabling Lint/MissingSuper

### ISSUE THREAD

**Issue by will** [testdouble/standard#195]
Hi, I just updated and noticed this linter was turned on in 0.5.0.

It has a ton of problems for me since I use [`abstract!` from sorbet](https://sorbet.org/docs/abstract). It generates an initialize method in the abstract parent class to raise an exception if anyone accidentally tries to instantiate it. All of the child classes have their own `initialize` methods and are correctly not calling `super`, but the linter thinks they should be calling it.

I'm going to disable it for myself in the yaml file, but wanted to open the issue here in case others have the same problem.

**Comment by searls** [comment]
Thanks for calling this out! Enough people use sorbet that I think this alone is worth rolling back. Even as I added it I could imagine a couple corner cases where not calling super was the right thing to do

**Comment by searls** [comment]
Landed in 0.5.2

**Comment by will** [comment]
Thanks! 

Also, rubocop may have disabled `Style/MethodMissingSuper` when they put in `Lint/MissingSuper ` https://github.com/rubocop-hq/rubocop/pull/8376

I do remember it yelling at me about the method missing one, but I cant remember if it was useful or not.

**Comment by searls** [comment]
IIRC the MissingSuper is a recent replacement for the other one

**Comment by will** [comment]
That's my understanding, but if the new MissingSuper is disabled here, you might want to turn back on MethodMissingSuper manually?

**Comment by searls** [comment]
My understanding is `MethodMissingSuper` was removed in favor of `MissingSuper` and there is unfortunately no apparent way to configure it to only cover method missing 😦 

https://docs.rubocop.org/rubocop/0.89/cops_lint.html

**Comment by marcandre** [comment]
> It has a ton of problems for me since I use [`abstract!` from sorbet](https://sorbet.org/docs/abstract). It generates an initialize method in the abstract parent class to raise an exception if anyone accidentally tries to instantiate it.

Interesting. I couldn't read this in the reference given, do you know of another source?

Unless I'm missing something, it seems completely mistaken from `sorbet`. They should instead override `MyAbstractClass.new` instead of `MyAbstractClass#initialize`. An abstract class could have very valid reason to do initializations, and a concrete class shouldn't have to create an empty `initialize` if it doesn't require initialization parameters.

**Comment by searls** [comment]
Will reopen and see if this is a sorbet-specific thing and they're open to changing it. If so and nobody else raises issues I'll turn it back on

**Comment by ttilberg** [comment]
@searls For what it's worth, I ran into this and was surprised outside of Sorbet a few months back: https://www.reddit.com/r/ruby/comments/hw8t6a/how_to_please_standardrbrubocop/. It also references another user in Rubocops tracker with a similar issue: https://github.com/rubocop-hq/rubocop/issues/3760#issuecomment-353103957

**Comment by marcandre** [comment]
@ttilberg I agree, the rule doesn't quit work for `method_missing` and for `repond_to_missing?`; it sometimes makes sense to unconditionally return from those without calling `super`. I'll fix that later today 👍

**Comment by searls** [comment]
@ttilberg if you look at @marcandre's [PR](https://github.com/rubocop-hq/rubocop/pull/9072), the feedback received was to document why this hook should be an exception. Since you've written about this before and care a lot about it, could you provide a documentation update for inclusion in that PR so that it can be merged upstream?

**Comment by marcandre** [comment]
No worried, I'm on it
:-)

**Comment by jmkoni** [comment]
@will We are looking at turning this on because it's good for the average Rubyist. Has Sorbet changed? Or would this still be an issue?

**Comment by will** [comment]
@jmkoni do you know of a way that I can test turning it back on in a single project, to see what happens? Something in that yaml file maybe?

**Comment by jmkoni** [comment]
@will yeah! You should be able to do something like this, but instead enable `Lint/MissingSuper`. I'm guessing you already have a yaml file that looks very similar. https://github.com/testdouble/standard/issues/158#issuecomment-640118656

**Comment by will** [comment]
with a `.standard.yml` file of 

```yml
parallel: true
ignore:
  - 'sorbet/**/*'
  - '**/*':
    - Lint/ConstantDefinitionInBlock # sorbet enums

Lint/MissingSuper:
  Enabled: true
```

standardrb 0.11.0 doesn't report any problems, however I would have expected these two classes

``` ruby
# typed: strict
# frozen_string_literal: true

class Factory
  extend T::Sig
  extend T::Helpers
  abstract!

  # sorbet workaround, ref: https://github.com/sorbet/sorbet/issues/2374
  alias_method :initialize_abstract, :initialize

  sig { params(params: T::Hash[Symbol, T.untyped]).void }
  def initialize(params = {})
    initialize_abstract
  end
end

class ClusterFactory < Factory
  extend T::Sig

  sig { params(params: T::Hash[Symbol, T.untyped]).void }
  def initialize(params = {})
    # no super
  end
end
```

to have generated an error with this on?

**Comment by will** [comment]
Oh I see it was supposed to be in `.rubocop.yml` not standard, and I was supposed to run rubocop directly and not standard. 

This still breaks for me in that case above. Also in another area I'm on purpose not calling super in subclasses, but this is not related to anything sorbet. 

Either way I'm okay with whatever you all decide as long as I can still opt out of this check with an ignore, which I think would work like I'm doing for `Lint/ConstantDefinitionInBlock`

**Comment by searls** [comment]
TIL (well, re-learned) that we allowed rule-specific ignores in Standard. That being the case I think we should turn this back on and hope Sorbet users who need this will find this issue.

**Comment by marcandre** [comment]
> hope Sorbet users who need this will find this issue

and upvote / comment on the issue I opened on Sorbet...

**Comment by jasonkarns** [comment]
This issue was originally intending to disable the `Lint/MissingSuper` rule, thanks to sorbet. It was fixed: https://github.com/standardrb/standard/issues/195#issuecomment-680866452

Then there was a request to change sorbet's upstream behavior
> Edit: I opened [sorbet/sorbet#3388](https://github.com/sorbet/sorbet/issues/3388)

Which was resolved: https://github.com/sorbet/sorbet/pull/6888

So the expectation was that this rule would be _re-enabled_: https://github.com/standardrb/standard/issues/195#issuecomment-683786872

There was also a change to rubocop's rule to have it exclude a subset (and thus make the rule more useful+safe when enabled): https://github.com/rubocop/rubocop/pull/9072


The change in sorbet and rubocop makes it now desirable to have this rule re-enabled, which this issue was repurposed to request: https://github.com/standardrb/standard/issues/195#issuecomment-766090038

I don't know when, but github has either broken (or removed) the rendering of markdown in PR titles. To that end, I'm removing the bit of the title that _used_ to be strike-through, because it's confusing. But the desire now is to _re-enable_ the rule.

**Comment by searls** [comment]
:+1: to Jason. Same read
